daemon/cmd/daemon_main.go

```diff
@@ -1603,7 +1600,7 @@ func newDaemonPromise(params daemonParams) (promise.Promise[*Daemon], legacy.Dae
 // startDaemon starts the old unmodular part of the cilium-agent.
 // option.Config has already been exposed via *option.DaemonConfig promise,
 // so it may not be modified here
-func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *daemonCleanup, params daemonParams) error {
+func startDaemon(d *Daemon, restoredEndpoints *restoration.EndpointRestoreState, cleaner *daemonCleanup, params daemonParams) error {
```

Comment:
Let's get rid of endpoint restoration + regeneration entirely from the daemon. Instead, the EndpointManager or a sub-Cell should restore from disk, provide the endpoints to any cells that need them, then push them in to the endpoint manager and trigger regeneration.

Then, the FQDN bootstrapper could call some function like `RegisterEndpointRestoreCallback()` or something that then synchronously calls back from the restorer to the FQDN bootstrapper at the right time.